### PR TITLE
Avoid generating SemiJoins when the subquery contains an OFFSET expression, as doing say may generate incorrect results.

### PR DESCRIPTION
**Pull request by nicktobey**

### CODE REVIEW

**Comment by max-hoffman** [comment]
Re the targeted test:
```
SemiJoin
 ├─ AND
 │   ├─ GreaterThanOrEqual
 │   │   ├─ vals.val:0
 │   │   └─ ranges.min:1
 │   └─ LessThanOrEqual
 │       ├─ vals.val:0
 │       └─ ranges.max:2
 ├─ ProcessTable
 │   └─ Table
 │       ├─ name: vals
 │       └─ columns: [val]
 └─ Offset(1)
     └─ Table 
     ├─ name: ranges
     └─ columns: [min max]
```

I would have expected the problem here to be dropping the Limit, not the decorrelation necessarily. Do we always drop limit nodes when we decorrelate?

**Comment by nicktobey** [comment]
I think we do drop the limit on purpose, because the SemiJoin produces ExistsIter. We don't care about how many results are in the subqeury, just that there are results.

**Review by max-hoffman — APPROVED** [review]
LGTM, only suggestion is maybe adding a test with correctness in addition to join existence

**Comment by max-hoffman** [comment]
> I think we do drop the limit on purpose, because the SemiJoin produces ExistsIter. We don't care about how many results are in the subqeury, just that there are results.

That makes sense! Thanks for fixing

File: enginetest/join_planning_tests.go [hunk 1]
```diff
@@ -1298,20 +1298,17 @@ join uv d on d.u = c.x`,
...
-							{1},
-							{2},
-							{3},
-							{4},
-							{5},
-						},
-					},
-			*/
+			{
+				q:     "select * from vals where exists (select * from ranges where val between min and max limit 1 offset 1);",
```
[the remark this answers is not in the document]
Comment (nicktobey) [reply enginetest/join_planning_tests.go:1302]: Discussed offline. The current test already accomplishes this.

File: enginetest/join_planning_tests.go [hunk 2]
```diff
@@ -1672,6 +1669,8 @@ func TestJoinPlanning(t *testing.T, harness Harness) {
 		t.Run(tt.name, func(t *testing.T) {
 			harness.Setup([]setup.SetupScript{setup.MydbData[0], tt.setup})
 			e := mustNewEngine(t, harness)
+			// e.EngineAnalyzer().Debug = true
+			// e.EngineAnalyzer().Verbose = true
```
[the remark this answers is not in the document]
Comment (nicktobey) [reply enginetest/join_planning_tests.go]: Fixed.

**Pull request merged**